### ISSUE TITLE
Make regulization parameter more physical

### ISSUE THREAD

**Issue by kapytaine** [statsmodels/statsmodels#8955]
#### Is your feature request related to a problem? Please describe
In the `GLM` class, I don't understand why, when fitting a regularized model, the loss function is `-(self.loglike(x) / self.nobs - np.sum(alpha * x**2) / 2)` (loss from `_fit_ridge` method). Considering this formula, `alpha` includes both the variance (that I note `sigma2`) of the parameter prior distribution, which is a normal one for Ridge, and the number of the observations (that I note `n_obs`): `alpha = 1/(n_obs*sigma2)`. From the mathematical standpoint, I expect the regularization to have less and less weight when increasing the sample size since the posterior distribution is proportional to the likelihood times the prior distribution (and the likelihood scales with the number of observations) (see [wikipedia](https://en.wikipedia.org/wiki/Posterior_probability#Calculation)). For instance, this is implemented with this logic in the [ridge regression of scikit-learn](https://scikit-learn.org/stable/modules/linear_model.html#regression) (even though there are other models in scikit-learn with the other approach). 

#### Describe the solution you'd like
Considering the above explanation, why didn't you envisage another loss:
- `-(self.loglike(x) - np.sum(alpha * x**2) / 2)`, or 
- `-(self.loglike(x) - np.sum(alpha * x**2) / 2)  / self.nobs` (if the division is required)

`alpha` could then be interpreted as the inverse of the variance of the parameter prior distribution `alpha = 1/sigma2`. For a fixed `alpha`, the regularization would have a large weight if there is few observations and its weight would reduce while increasing the sample size. Nevertheless, I guess there is reason being the current implementation. So the solution I propose is perhaps more a question: why did you consider this loss which makes `alpha = 1/(n_obs*sigma2)`? I personally prefer my proposition which seems more logic (to me) even though I know there are the 2 formulations in statistical packages.

#### Describe alternatives you have considered
If this is done on purpose, don't you think a note may be needed to explain:
- the reason of this trade-off between likelihood and prior, and
- the meaning of `alpha`?

#### Additional context
